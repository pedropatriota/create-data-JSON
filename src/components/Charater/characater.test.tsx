import { cleanup, render, screen } from "@testing-library/react";
import Character from ".";
import { highlightMatchedString } from ".";

const setup = () => {
  const utils = render(
    <Character
      inputValue="Rick"
      name={"Rick Sanchez"}
      image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
  );

  return {
    ...utils,
  };
};

afterEach(cleanup);

describe("ListItem", () => {
  it("should render each item from the list", async () => {
    setup();

    const image = await screen.findByAltText("Rick Sanchez");
    const name = await screen.findByText("Rick");

    expect(image).toBeTruthy();
    expect(name).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});

describe("highlightMatchedString", () => {
  it("should return the original text when there is no match", () => {
    const text = "Hello";
    const highlight = "notfound";
    const result = highlightMatchedString(text, highlight);
    const { container } = render(result);
    expect(container.innerHTML).toBe("<span>Hello</span>");
  });

  it("should highlight the matched part of the string", () => {
    const text = "Hello, World!";
    const highlight = "hello";
    const result = highlightMatchedString(text, highlight);
    const { container } = render(result);
    expect(container.innerHTML).toBe(
      "<span><strong>Hello</strong>, World!</span>"
    );
  });

  it("should be case-insensitive when highlighting", () => {
    const text = "apple orange apple";
    const highlight = "APPLE";
    const result = highlightMatchedString(text, highlight);
    const { container } = render(result);
    expect(container.innerHTML).toBe(
      "<span><strong>apple</strong> orange <strong>apple</strong></span>"
    );
  });
});
