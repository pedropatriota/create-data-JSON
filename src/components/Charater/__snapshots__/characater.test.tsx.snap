// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<div>
  <li
    class="sc-aXZVg dXIGHn"
  >
    <div
      class="sc-gEvEer kjrfHW"
    >
      <img
        alt="Rick Sanchez"
        class="sc-eqUAAy jqhXwM"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <span>
        <strong>
          Rick
        </strong>
         Sanchez
      </span>
    </div>
  </li>
</div>
`;
